File: Algorithms/Maths/Magic Number/Magic Number.py
input_num = int(input())

def isMagic(input_num):
	z = len(list(str(input_num)))
	for x in range(1,z+1):
		rem = int(input_num % 10)
		input_num = int(input_num/10)
		input_num = rem + input_num
	if(input_num == 1):
		print('Magic Number')
	else:
		print("Not a Magic Number")


isMagic(input_num)

File: Algorithms/Maths/Magic Number/Nodejs/MagicNumber.js

const readline = require("readline");

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

rl.question("Enter the  number: ", a => {
    var n=parseInt(a)
    var length= a.length;
    while(length>1){
        var arr = n.toString(10).replace(/\D/g, '0').split('').map(Number);


    n = arr.reduce(myFunc);
    length=n.toString().length;
    }
    if(n==1)
    console.log("The number is a magic number");
    else
    console.log("The number is not a magic number");
    process.exit();
  });


function myFunc(total, num) {
    return total + num;
}